src/pcre2_compile.c

```diff
@@ -1113,7 +1112,7 @@ in the decoded tables. */
 
 if ((code->flags & PCRE2_DEREF_TABLES) != 0)
   {
-  ref_count = (PCRE2_SIZE *)(code->tables + TABLES_LENGTH);
+  PCRE2_SIZE *ref_count = (PCRE2_SIZE *)(code->tables + TABLES_LENGTH);
```

Comment:
Personally, I'm very happy with these changes.

I know Philip likes the old style of defining variables high up, at the top of a scope, and with a blank line after variable definitions.

But I don't see any benefit to having variables available for use, but not yet initialised. Much better to define & initialise at the same time (safer).

The compiler will hoist all the variables up to the top anyway (it will bump the stack pointer just once at the start of a block, rather than bump the stack pointer multiple times, when it sees a new variable).

Reply:
Partly it's because I'm a dinosaur from the age when one had to define variables like that, but partly also I find it makes it easier when looking back up some code to find where a variable is defined. However, I am not going to try to impose my own preferences on the future. I can certainly see the advantage of always initializing at definition time. So please don't worry about me too much.

Reply:
Funny is that this change is still valid C89 code and the main motivation wasn't to go against Philip's advice of defining variables at the beginning of blocks, but just reducing the scope of this variable to where it was actually needed/used.

Since we have at least one CI job with `-Wshadow` and I wanted to minimize churn didn't rename the variable to reflect its "temp" holder (might be even optimized out) status.